aptos-move/dynamic-transaction-composer/src/codegen.rs

```diff
@@ -0,0 +1,380 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    ArgumentOperation, BatchArgument, BatchedFunctionCall, PreviousResult, APTOS_SCRIPT_BUILDER_KEY,
+};
+use move_binary_format::{
+    access::{ModuleAccess, ScriptAccess},
+    builders::CompiledScriptBuilder,
+    errors::{PartialVMError, PartialVMResult},
+    file_format::*,
+};
+use move_core_types::{
+    identifier::IdentStr,
+    language_storage::{ModuleId, TypeTag},
+    metadata::Metadata,
+    transaction_argument::TransactionArgument,
+    vm_status::StatusCode,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Default)]
+struct Context {
+    script: CompiledScript,
+    // Arguments to the compiled script.
+    args: Vec<Vec<u8>>,
```

Comment:
I don't quite understand what this here is. The BCS of a set of concrete arguments? Why would this be on this level?

I would the compiled script expected to be parametric over a set of formal parameters, so it can be stored and executed many times with different parameters.